lib/AST/Decl.cpp

```diff
@@ -3612,6 +3612,10 @@ void VarDecl::emitLetToVarNoteIfSimple(DeclContext *UseDC) const {
     // If the problematic decl is 'self', then we might be trying to mutate
     // a property in a non-mutating method.
     auto FD = dyn_cast_or_null<FuncDecl>(UseDC->getInnermostMethodContext());
+
+    // If this is a subscript getter, don't suggest adding mutating.
+    if (FD->getAccessorStorageDecl()->getGetter() == FD) return;
```

Comment:
I figured out how to detect the implicit get block:
```
(FD->getAccessorStorageDecl()->getBracesRange().Start == FD->getLoc())
```

The fixit is turning out to be a bit trickier though and I can't find any precedent to copy. I'll keep looking, but if you can think of one that would help.